components/org.wso2.bfsi.consent.management.common/src/main/java/org/wso2/bfsi/consent/management/common/caching/ConsentManagementBaseCache.java

```diff
@@ -0,0 +1,212 @@
+                        key.toString().replaceAll("[\r\n]", ""), cacheName.replaceAll("[\r\n]", "")));
+            }
+
+            removeFromCache(key);
+            addToCache(key, value);
+            return value;
+        }
+
+    }
+
+    /**
+     * Get from cache.
+     *
+     * @param key cache key.
+     * @return cached object.
+     */
+    public V getFromCache(K key) {
+
+        Cache<K, V> cache = getBaseCache();
+
+        if (cache.containsKey(key)) {
+
+            if (log.isDebugEnabled()) {
+                log.debug(String.format("Found cache entry `%s` in cache %s",
+                        key.toString().replaceAll("[\r\n]", ""), cacheName.replaceAll("[\r\n]", "")));
+            }
+            return cache.get(key);
+        } else {
+
+            return null;
+        }
+
+    }
+
+    /**
+     * Add Object to cache.
+     *
+     * @param key    cache key.
+     * @param value  cache value.
+     */
+    public void addToCache(K key, V value) {
+
+        if (log.isDebugEnabled()) {
+            log.debug(String.format("`%s` added into cache %s", key.toString().replaceAll("[\r\n]", ""),
+                    this.cacheName.replaceAll("[\r\n]", "")));
+        }
+
+        Cache<K, V> cache = getBaseCache();
+        cache.put(key, value);
+    }
+
+    /**
+     * Remove Object from Cache.
+     *
+     * @param key  cache key.
+     */
+    public void removeFromCache(K key) {
+
+        if (log.isDebugEnabled()) {
+            log.debug(String.format("`%s` removed from cache %s", key.toString().replaceAll("[\r\n]", ""),
+                    cacheName.replaceAll("[\r\n]", "")));
+        }
+
+        Cache<K, V> cache = getBaseCache();
+        cache.remove(key);
+    }
+
+    /**
+     * Get Cache for instance.
+     *
+     * @return
+     */
+    private Cache<K, V> getBaseCache() {
+
+        CacheManager cacheManager = Caching.getCacheManager(BASE_CACHE_KEY);
+
+        Iterable<Cache<?, ?>> availableCaches = cacheManager.getCaches();
+        for (Cache cache : availableCaches) {
+            if (cache.getName().equals(cache.getName().startsWith("$__local__$.") ? "$__local__$." +
+                    cacheName : cacheName)) {
+                return cacheManager.getCache(cacheName);
+            }
+        }
+
+        CacheConfiguration.Duration accessExpiry = new CacheConfiguration.Duration(TimeUnit.MINUTES,
+                getCacheAccessExpiryMinutes());
+
+        CacheConfiguration.Duration modifiedExpiry = new CacheConfiguration.Duration(TimeUnit.MINUTES,
+                getCacheModifiedExpiryMinutes());
+
+        // Build Cache on OB base cache.
+        CacheBuilder<K, V> cacheBuilder = cacheManager.createCacheBuilder(cacheName);
+
+        return cacheBuilder.setExpiry(CacheConfiguration.ExpiryType.ACCESSED, accessExpiry)
+                .setExpiry(CacheConfiguration.ExpiryType.MODIFIED, modifiedExpiry)
+                .build();
+
```

Comment:
```suggestion
```

Reply:
Fixed in https://github.com/wso2-extensions/bfsi-consent-management/pull/5/commits/4d9138c332549a29ab817615c49fd31c6d8cc121